ui/src/components/EntityModal/EntityModal.test.tsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import EntityModal from './EntityModal';
+import { setUnifiedConfig } from '../../util/util';
+import {
+    EverythingConfig,
+    configBasicOauthDisableonEdit,
+    configOauthOauthDisableonEdit,
+} from './TestConfig';
+
+describe('EntityModal - Basic oauth', () => {
+    const handleRequestClose = jest.fn();
+
+    const setUpConfigWithDisabedBasicOauth = () => {
+        const newConfig = EverythingConfig;
+        // incompatibile options types, TODO implement global type for config
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        newConfig.pages.configuration.tabs = [configBasicOauthDisableonEdit as any];
+
+        setUnifiedConfig(newConfig);
+    };
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const renderModalWithProps = (props: any) => {
+        render(<EntityModal {...props} handleRequestClose={handleRequestClose} />);
+    };
+
+    const getDisabledBasicField = () =>
+        document.getElementsByClassName('basic_oauth_text_jest_test')[1];
+
+    it('if oauth field not disabled with create after disableonEdit true', async () => {
+        setUpConfigWithDisabedBasicOauth();
+        const props = {
```

Comment:
```suggestion
        const props: EntityModalProps = {
```

Reply:
since the props are declared in function param, here we do not need to duplicate prop types, wdyt?

Reply:
Indeed, we need `satisfies EntityModalProps;` Good catch
<img width="851" alt="image" src="https://github.com/splunk/addonfactory-ucc-generator/assets/142901247/f17f48e1-88a9-4db9-a1c1-d6b0c67ef390">